### ISSUE TITLE
Very different results for same dataset during training and using predict function

### ISSUE THREAD

**Issue by ririya** [keras-team/keras#12952]
**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution - Linux Ubuntu 18.04:  
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.7.0
- Keras version:  2.2.2
- Python version:  3.6.8 |Anaconda
- CUDA/cuDNN version:  cuda toolkit 9.0, cuDNN 7.1.2
- GPU model and memory: 2x GTX 1080 Ti (22 gb)

**Describe the current behavior**  

I have a video classification problem with 2 classes and I'm using a Resnet50 + LSTM architecture.

During training, the accuracy result of each mini-batch is pretty consistent and close to 100%. However after the network is already trained if I test the training dataset with the predict or predict_generator functions I get 100% accuracy for one of the classes and 65 /35 % for the other.

This behavior happens for some datasets. For some datasets, the network behaves as expected, for others this strange behavior occurs and I have not been able to identify the cause of the problem.

**Describe the expected behavior**  

Results from training feedback and predict function should match

**Code to reproduce the issue**  

```
def single_class_accuracy(class_id):
    def fn(y_true, y_pred):
        class_id_true = K.argmax(y_true, axis=-1)
        class_id_preds = K.argmax(y_pred, axis=-1)
        accuracy_mask = K.cast(K.equal(class_id_true, class_id), K.floatx())
        class_acc_tensor = K.cast(K.equal(class_id_true, class_id_preds), K.floatx()) * accuracy_mask
        class_acc = K.sum(class_acc_tensor) / K.maximum(K.sum(accuracy_mask), 1)
        return class_acc
    return fn

def conf_matrix_diag_acc(numClasses, batch_size):
    def fn_conf_matrix_diag_acc(y_true, y_pred):

        val = np.zeros((batch_size,numClasses))
        class_acc_total = K.variable(value=val, dtype='float32')

        for c in range(numClasses):

            curr_fn = single_class_accuracy(c)
            class_acc = curr_fn(y_true,y_pred)
            class_acc_total = class_acc_total + class_acc

        return class_acc_total/numClasses
    return fn_conf_matrix_diag_acc`
```

```
def CNN_LSTM_2(params):

    if params.makeRGB:
        input_shape = (params.numFrames, params.input_height, params.input_width, 3)
    else:
        input_shape = (params.numFrames, params.input_height, params.input_width, 1)



    if params.Base_Model_Type == 'VGG16':
        baseModel = vgg16.VGG16(include_top=False, weights=params.preTrainedWeights, pooling='avg')
    elif params.Base_Model_Type == 'InceptionV3':
        baseModel = inception_v3.InceptionV3(include_top=False, weights='imagenet', pooling='avg')

    elif params.Base_Model_Type == 'ResNet50':
        baseModel = resnet50.ResNet50(include_top=False, weights=params.preTrainedWeights, pooling='avg')

    elif params.Base_Model_Type == 'FC':
        baseModel = fullyConnectedBase(params)
        # baseModel = create_base(params)


    input_layer = Input(shape=input_shape)
    input_mask = Input(shape=(params.numFrames,1))

    curr_layer = TimeDistributed(baseModel)(input_layer)

    # curr_layer = Reshape(target_shape=(params.numFrames, 2048))(curr_layer)
    # curr_layer = TimeDistributed(Flatten())(curr_layer)

    resnetOutput = Dropout(0.5)(curr_layer)

    curr_layer = resnetOutput

    # curr_layer = multiply([resnetOutput,input_mask])

    # cnn_output = curr_layer

    # curr_layer = Masking(mask_value=0.0)(curr_layer)



    lstm_out = LSTM(128, return_sequences=True, dropout=0.5)(curr_layer)

    all_lstm_outs = []

    for outInd in range(params.numLSTMOutputs):
        seqInd =  (outInd+1)*int(params.numFrames/params.numLSTMOutputs) -1
        currIn = Lambda(lambda x: slice(x, (0,seqInd, 0), (-1,1, -1)))(lstm_out)

        all_lstm_outs.append(currIn)

    concat = concatenate(all_lstm_outs)

    output = Dense(output_dim=params.numClasses, activation='softmax')(concat)
    output = Reshape((params.numClasses,), name='output')(output)

    model = Model([input_layer, input_mask], output)


    plot_model(model, show_shapes=True, to_file='cnn_lstm_2_model.png')

    return model
`
```

**Comment by channingxiao** [comment]
I had same problem with a very simple task. see https://github.com/keras-team/keras/issues/12953, could you verify what I found.

**Comment by ririya** [comment]
> I had same problem with a very simple task. see #12953, could you verify what I found.

Hi channingxiao thanks for replying!

Did this solve your issue? I noticed the training parameter is evaluated only during network construction time, so it is not called again during training, I'm not sure how this is supposed to be used. 

I also tried changing **"def call(self, inputs, training=None):" at normalization.py line 132
to "def call(self, inputs)"** but then I get errors because the training parameter is not found. How exactly did you solve this?
